src/corecel/data/Copier.cc

```diff
@@ -11,18 +11,20 @@
 
 #include "corecel/device_runtime_api.h"
 #include "corecel/Macros.hh"
+#include "corecel/sys/Device.hh"
+#include "corecel/sys/Stream.hh"
 
 namespace celeritas
 {
-//---------------------------------------------------------------------------//
-/*!
- * Perform a memcpy on the data.
- */
-void copy_bytes(MemSpace dstmem,
-                void* dst,
-                MemSpace srcmem,
-                void const* src,
-                std::size_t count)
+namespace
+{
+template<bool async>
+inline void copy_bytes_impl(MemSpace dstmem,
+                            void* dst,
+                            MemSpace srcmem,
+                            void const* src,
+                            std::size_t count,
+                            [[maybe_unused]] StreamId stream)
```

Comment:
This will probably need another GCC 8-9 workaround... rather than using the `template<bool async>` I think it'd be better to have a standalone function `decltype(auto) to_memcpy_kind(MemSpace src, MemSpace dst)` that you reuse in the separate sync/async functions.